trl/trainer/utils.py

```diff
@@ -881,7 +881,8 @@ def print_rich_table(df: pd.DataFrame) -> Table:
 SIMPLE_SFT_CHAT_TEMPLATE = "{% for message in messages %}{{' ' + message['content']}}{% endfor %}{{ eos_token }}"
 # SIMPLE_SFT_CHAT_TEMPLATE simply ends things with an EOS token, this helps the SFT model learn to end the completions with EOS tokens
 
-SIMPLE_QUERY_CHAT_TEMPLATE = "{% for message in messages %}{{' ' + message['content']}}{% endfor %}"
+SIMPLE_QUERY_CHAT_TEMPLATE = "{% for message in messages %}{{message['role'] + ': ' + message['content'] + '\n\n'}}{% endfor %}{% if add_generation_prompt %}{{ 'assistant:' }}{% endif %}"
```

Comment:
Would a good default be the one from Anthropic, which has alternating `Human:` and `Assistant:` roles? https://huggingface.co/datasets/Anthropic/hh-rlhf

Reply:
Something like this:?

```suggestion
SIMPLE_QUERY_CHAT_TEMPLATE = "{% for message in messages %}{% if message['role'] == 'user' %}Human: {% elif message['role'] == 'assistant' %}Assistant: {% endif %}{{ message['content'] }} \n\n{% endfor %}{% if add_generation_prompt %}Assistant:{% endif %}"
```

```python
>>> from transformers import AutoTokenizer
>>> tokenizer = AutoTokenizer.from_pretrained("gpt2")
>>> tokenizer.chat_template = "{% for message in messages %}{% if message['role'] == 'user' %}Human: {% elif message['role'] == 'assistant' %}Assistant: {% endif %}{{ message['content'] }} \n\n{% endfor %}{% if add_generation_prompt %}Assistant:{% endif %}"
>>> messages = [
...     {"role": "user", "content": "Hello, how are you?"},
...     {"role": "assistant", "content": "I'm doing great. How can I help you today?"},
...     {"role": "user", "content": "I'd like to show off how chat templating works!"},
... ]
>>> print(tokenizer.apply_chat_template(messages, tokenize=False))
Human: Hello, how are you? 

Assistant: I'm doing great. How can I help you today? 

Human: I'd like to show off how chat templating works! 
```

Reply:
Let's just capitalise the role, so that it keep things simple (we don't really need the user to be converted to human))

```suggestion
SIMPLE_QUERY_CHAT_TEMPLATE = "{% for message in messages %}{{message['role'].capitalize() + ': ' + message['content'] + '\n\n'}}{% endfor %}{% if add_generation_prompt %}{{ 'Assistant:' }}{% endif %}"
```